alacritty_terminal/src/term/mod.rs

```diff
@@ -1794,8 +2167,18 @@ impl<T: EventListener> ansi::Handler for Term<T> {
         let source = if self.alt { &self.cursor_save_alt } else { &self.cursor_save };
 
         self.cursor = *source;
-        self.cursor.point.line = min(self.cursor.point.line, self.grid.num_lines() - 1);
-        self.cursor.point.col = min(self.cursor.point.col, self.grid.num_cols() - 1);
+        let line = min(self.cursor.point.line, self.grid.num_lines() - 1);
+        let col = min(self.cursor.point.col, self.grid.num_cols() - 1);
+
+        self.damage = DamageRect::bounding_rect(&self.damage, &DamageRect {
+            x: min(self.cursor.point.col.0, col.0),
+            y: min(self.cursor.point.line.0, line.0),
```

Comment:
You don't need this `min` since you've already computed them a line above, since it computes min based on .0 field already.